atsign-foundation/at_client_sdk
# fix: Enhance the metadata to retain existing metadata or unset the metadata

**-What I did**
* Enhance the metadata as follows:
  * When an attribute is set to a different value than the existing one, overwrite with the new value
  * When an attribute is not set, then fetch the existing metadata and add it to the current metadata
  * When an attribute is explicitly set to null, reset the metadata.

**-How I did**
* In LocalSecondary, introduce a method "_unsetOrRetainMetadata", that has the implementation for the above behaviour.
*  Refactoring: Move the code that sets the metadata in the "executeVerb" method to a new method "_getAtMetadataFromVerbBuilder".

**-How to verify it**
* Added the following unit tests that assert the behaviour
  * A test to verify existing metadata is retained when new metadata is set
  * A test to verify new value is set to the old value
  * A test to verify setting null reset the metadata

**-Description for CHANGELOG**
* fix: Enhance the metadata to retain existing metadata or unset the metadata
